### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by Abe1able**
### In this milestone of the project, I have

- **created a method called current_user in the ApplicationController to make current user data available to all controllers.**

- **created forms to perform the following functions:**

> Create a Post on behalf of the current_user
> Create a comment on behalf of the current_user
> Allow Users to add likes to Posts.

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1,2 +1,13 @@
...
+			<div class="img-cont">
+				<%# <%= image_tag user.photo, :size => "50x50", alt: "user photo" %> %>
+			</div>
+			<div class="user-details">
+				<h2><%= user.name %></h2>
+				<p>Number of posts: <%= user.posts_counter %></p>
+			</div>
+		</div>
+	<%end%>
+</div>
```
Comment (zieeco) [inline app/views/users/index.html.erb:13]: - Hey, the index page for your users doesn't seem to show up anything, it's blank even after I seed your database with some data. please, try to fix this then you can add the form functions properly. Please, ensure all your views pages are working correctly then you can proceed to add the form interaction.


image

 - For instance, I add a user with the id of 1, it throws an error. please, try to fix your controllers so you can implement the form functionality.

image

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @Abe1able,




Great job so far! 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- Nice trying to add the form ✔️
- Good commit message. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._


## Optional suggestions

- N/A


**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/users/index.html.erb (hunk 1 above)
Comment (Abe1able) [reply app/views/users/index.html.erb:13]: I've made sure it works correctly. Here is the screenshot.

Bloggingly4

**Review by afanimartin — APPROVED** [review]
Hi @Abe1able, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Changes made as requested by the previous reviewer

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

## Find me on slack
### Martin Afani

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**